### PR TITLE
feat/connector-elsys

### PR DESCRIPTION
**Pull request by mateuscardosodeveloper**
## Decoder Description

Add ELSYS sensor decoders

## Type of change

- [x] Adding a new Decoder of Connector
- [ ] Adding a new Decoder of Network
- [ ] Update or fixing an issue in existing Decoder

## Decoder Information and Payload to test and review

- [x] Documentation of the hardware or protocol:
- [ ] Payload of example the test the decoder:

## Checklist for Adding a New Decoder

- [ ] Created a new folder under `./decoders/network/` or `./decoders/connector/` with the name of your decoder.
- [ ] Added a `network.jsonc` or `connector.jsonc` file that follows the structure defined in `./schema/`.
- [ ] Created version folders and added `manifest.jsonc` files for each version.
- [x] Followed the folder structure guidelines for manufacturer and sensor/device model.
- [ ] The code has unit test and it's in TypeScript.

### CODE REVIEW

**Review by Freddyminu — APPROVED** [review]

File: decoders/connector/elsys/ems/v1.0.0/payload.js [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+function Decoder(bytes, port) {
+  return DecodeElsysPayload(bytes);
+}
+
+
+// payload = [{ variable: 'payload', value: '0100e202290400270506060308070d62', serie: '122a' }];
+
+const data = payload.find(x => x.variable === 'data' || x.variable === 'payload');
+if (data) {
+  const serie = data.serie || new Date().getTime();
```
Comment (vitorfdl) [inline decoders/connector/elsys/ems/v1.0.0/payload.js:225]: serie is deprecated. This payload parser must be updated to use group instead

File: decoders/connector/elsys/ers-co2/v1.0.0/payload.js [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+function toTagoFormat(object_item, serie, prefix = '', location_var) {
+  const result = [];
+  for (const key in object_item) {
+    // if (ignore_vars.includes(key)) continue;
+
+    if (typeof object_item[key] === 'object') {
+      result.push({
+        variable: object_item[key].variable || `${prefix}${key}`,
+        value: object_item[key].value,
+        serie: object_item[key].serie || serie,
```
Comment (vitorfdl) [inline decoders/connector/elsys/ers-co2/v1.0.0/payload.js:39]: serie is deprecated. This payload parser must be updated to use group instead

File: decoders/connector/elsys/ers-co2/v1.0.0/payload.js [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+function Decoder(bytes, port) {
+  return DecodeElsysPayload(bytes);
+}
+
+
+// payload = [{ variable: 'payload', value: '0100e202290400270506060308070d62', serie: '122a' }];
+
+const data = payload.find(x => x.variable === 'data' || x.variable === 'payload');
+if (data) {
+  const serie = data.serie || new Date().getTime();
```
Comment (vitorfdl) [inline decoders/connector/elsys/ers-co2/v1.0.0/payload.js:225]: serie is deprecated. This payload parser must be updated to use group instead

File: decoders/connector/elsys/ers-desk/v1.0.0/payload.js [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+function Decoder(bytes, port) {
+  return DecodeElsysPayload(bytes);
+}
+
+
+// payload = [{ variable: 'payload', value: '0100e202290400270506060308070d62', serie: '122a' }];
+
+const data = payload.find(x => x.variable === 'data' || x.variable === 'payload');
+if (data) {
+  const serie = data.serie || new Date().getTime();
```
Comment (vitorfdl) [inline decoders/connector/elsys/ers-desk/v1.0.0/payload.js:225]: serie is deprecated. This payload parser must be updated to use group instead

File: decoders/connector/elsys/ers-eye/v1.0.0/payload.js [hunk 5]
```diff
@@ -0,0 +1,230 @@
...
+function Decoder(bytes, port) {
+  return DecodeElsysPayload(bytes);
+}
+
+
+// payload = [{ variable: 'payload', value: '0100e202290400270506060308070d62', serie: '122a' }];
+
+const data = payload.find(x => x.variable === 'data' || x.variable === 'payload');
+if (data) {
+  const serie = data.serie || new Date().getTime();
```
Comment (vitorfdl) [inline decoders/connector/elsys/ers-eye/v1.0.0/payload.js:225]: serie is deprecated. This payload parser must be updated to use group instead

File: decoders/connector/elsys/ers-lite/v1.0.0/payload.js [hunk 6]
```diff
@@ -0,0 +1,230 @@
...
+function Decoder(bytes, port) {
+  return DecodeElsysPayload(bytes);
+}
+
+
+// payload = [{ variable: 'payload', value: '0100e202290400270506060308070d62', serie: '122a' }];
+
+const data = payload.find(x => x.variable === 'data' || x.variable === 'payload');
+if (data) {
+  const serie = data.serie || new Date().getTime();
```
Comment (vitorfdl) [inline decoders/connector/elsys/ers-lite/v1.0.0/payload.js:225]: serie is deprecated. This payload parser must be updated to use group instead

File: decoders/connector/elsys/ers-sound/v1.0.0/payload.js [hunk 7]
```diff
@@ -0,0 +1,230 @@
...
+function Decoder(bytes, port) {
+  return DecodeElsysPayload(bytes);
+}
+
+
+// payload = [{ variable: 'payload', value: '0100e202290400270506060308070d62', serie: '122a' }];
+
+const data = payload.find(x => x.variable === 'data' || x.variable === 'payload');
+if (data) {
+  const serie = data.serie || new Date().getTime();
```
Comment (vitorfdl) [inline decoders/connector/elsys/ers-sound/v1.0.0/payload.js:225]: serie is deprecated. This payload parser must be updated to use group instead

File: decoders/connector/elsys/ers/v1.0.0/payload.js [hunk 8]
```diff
@@ -0,0 +1,233 @@
...
+function Decoder(bytes, port) {
+  return DecodeElsysPayload(bytes);
+}
+
+
+// payload = [{ variable: 'payload', value: '0100e202290400270506060308070d62', serie: '122a' }];
+
+const data = payload.find(x => x.variable === 'data' || x.variable === 'payload');
+if (data) {
+  const serie = data.serie || new Date().getTime();
```
Comment (vitorfdl) [inline decoders/connector/elsys/ers/v1.0.0/payload.js:228]: serie is deprecated. This payload parser must be updated to use group instead

**Review by vitorfdl — CHANGES_REQUESTED** [review]

**Comment by Freddyminu** [comment]
@mateuscardosodeveloper  Add the following tests: 

describe("Shall not be parsed", () => {
  let payload = [
    { variable: "shallnotpass", value: "04096113950292" },
    { variable: "fport", value: 9 },
  ];
  
  payload = decoderRun(file_path, { payload });
  
  test("Output Result", () => {
    expect(Array.isArray(payload)).toBe(true);
  });

  test("Not parsed Result", () => {
    expect(payload).toEqual([
      { variable: "shallnotpass", value: "04096113950292" },
      { variable: "fport", value: 9 },
    ]);
  });
});

**Review by vitorfdl — APPROVED** [review]

**Pull request merged**